src/entity/Post.ts

```diff
@@ -314,6 +314,10 @@ const mergeKeywords = async (
         })
         .filter((keyword) => !keyword.match(/^\d+$/)),
     );
+    // with the new system, a keyword needs to be added irrespective of a synonym
+    keywords.forEach((keyword) => {
```

Comment:
I think that we can simplify the process from line 307 to here.
What I suggest is to use the provided keywords array add to it the synonym that we get from the database, make sure everything is unique using `uniqueifyArray` and then filter using
`filter((keyword) => !keyword.match(/^\d+$/))`

So it should be roughly something like:
```
const additionalKeywords = synonymKeywords.map((synonym) => synonym.synonym)
const mergedKeywords = uniqueifyArray([...keywords, ...additionalKeywords].filter((keyword) => !keyword.match(/^\d+$/)))
```

I hope it makes sense. I didn't test the code

Reply:
Updated